### PR TITLE
vimPlugins.fzf-checkout-vim: init at 2021-07-21

### PR DESCRIPTION
**Pull request by viniciusmuller**
###### Motivation for this change
Add fzf-checkout plugin for managing git branches and tags with fzf.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130973` at ad8885c0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>spacevim</li>
    <li>tests.vim.test_vim_with_vim_nix</li>
    <li>vimPlugins.fzf-checkout-vim</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`configurePhase` should probably contain `runHook preConfigure` and `runHook postConfigure`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/misc/vim-plugins/build-vim-plugin.nix:28:14:
```
   |
28 |       inherit unpackPhase configurePhase buildPhase addonInfo preInstall postInstall;
   |              ^
```
</li>
  </ul>
</details>

File: pkgs/misc/vim-plugins/generated.nix [hunk 1]
```diff
@@ -1713,6 +1713,26 @@ final: prev:
...
+
+    # The plugin has a makefile which tries to run tests in a docker container.
+    # This prevents it.
+    prePatch = ''
+      rm Makefile
+    '';
+
+    meta.homepage = "https://github.com/stsewd/fzf-checkout.vim";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/misc/vim-plugins/generated.nix:1735]: That does not work and will be overwritten when regenerated. Please move the extra things to https://github.com/NixOS/nixpkgs/blob/master/pkgs/misc/vim-plugins/overrides.nix and generate the file.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by arcticlimer** [comment]
@SuperSandro2000 I think it should work now.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 130973 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>3 packages built:</summary>  <ul>    <li>spacevim</li>    <li>tests.vim.test_vim_with_vim_nix</li>    <li>vimPlugins.fzf-checkout-vim</li>  </ul></details>

**Pull request merged**